File: backend/app/schemas/document.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class DocumentResponse(BaseModel):
    id: int
    patient_id: int
    file_name: str
    file_type: str
    file_size: int
    uploaded_at: datetime
    extraction_status: str
    extraction_error: Optional[str] = None

    model_config = ConfigDict(from_attributes=True)










# from datetime import datetime

# from pydantic import BaseModel, ConfigDict

# class DocumentResponse(BaseModel):
#     id:int
#     patient_id:int
#     file_name:str
#     file_type:str
#     file_size:int
#     uploaded_at:datetime
    
#     model_config = ConfigDict(from_attributes=True)

    

File: frontend/src/types/document.ts
export type ExtractionStatus = "pending" | "processing" | "completed" | "failed";

export interface Document {
  id: number;
  patient_id: number;
  file_name: string;
  file_type: string;
  file_size: number;
  uploaded_at: string;
  extraction_status: ExtractionStatus;
  extraction_error?: string | null;
}

export interface ExtractedField {
  patient_document_data_id: number;
  patient_document_id: number;
  field_name: string;
  field_value: string | null;
  unit: string | null;
  reference_range: string | null;
  is_abnormal: boolean | null;
  display_order: number | null;
}









// export interface Document {
//     id:number;
//     patient_id:number;
//     file_name:number;
//     file_type:number;
//     file_size:number;
//     uploaded_at:number;
// }